tests/test_drain.py

```diff
@@ -0,0 +1,224 @@
+# Copyright (c) 2015 Uber Technologies, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+import pytest
+import tornado
+from tornado.concurrent import Future
+from tornado.iostream import StreamClosedError
+
+from tchannel import TChannel
+from tchannel import schemes
+from tchannel.errors import DeclinedError
+from tchannel.errors import NetworkError
+
+
+@pytest.mark.gen_test
+def test_stop():
+    server = TChannel("server")
+
+    @tornado.gen.coroutine
+    def endpoint(request):
+        raise tornado.gen.Return("hello")
+
+    server.raw.register("endpoint")(endpoint)
+    server.listen()
+    server.stop()
+
+    client = TChannel("client")
+    with pytest.raises(NetworkError):
```

Comment:
This would pass if we never started the server, need to assert something stronger.